### PR TITLE
513 make sure all filemappers uses mappingfilemapperbase get legacy value

### PR DESCRIPTION
**Pull request by fontanka16**

### CODE REVIEW

File: docs/mapping_file_based_mapping.md [hunk 1]
```diff
@@ -52,6 +52,18 @@ A typical entry looks like this:
 }
 ```
 folio_field and legacy_field are mandatory. All other fields are optional. 
+### The pirority of the mappings in the mapping entry:
+There might be exceptions to this in some areas, but generally, this is the how the mapping works:
+1. If there are reference data mappings or special cases for particular fields, then this has precedence
```
Comment (bltravis) [inline docs/mapping_file_based_mapping.md:57]: Perhaps:

> "Reference data mappings or other special cases are handled and the resulting value is returned"

File: docs/mapping_file_based_mapping.md [hunk 2]
```diff
@@ -52,6 +52,18 @@ A typical entry looks like this:
...
+2. Values added to the *value* field are returned immediately without any further manipulation
```
Comment (bltravis) [inline docs/mapping_file_based_mapping.md:58]: Does `value` override the reference data mapping? I don't think I've ever tried it…

Part of me feels like there should be some nesting with "or"s sprinkled throughout here…

File: docs/mapping_file_based_mapping.md [hunk 3]
```diff
@@ -52,6 +52,18 @@ A typical entry looks like this:
...
+3. Then, the *legacy field* value gets extracted from the source record.
+4. If there is an entry for rules.replaceValues, the extracted value is run through this process
+5. If there is an entry for rules.regexGetFirstMatchOrEmpty, the extracted value is run through this process as well and then return the value
```
Comment (bltravis) [inline docs/mapping_file_based_mapping.md:61]: Might re-word:

> "If there is an entry for `rules.regexGetFirstMatchOrEmpty`, the extracted value is run through this process and the result returned as the value"

File: docs/mapping_file_based_mapping.md [hunk 4]
```diff
@@ -52,6 +52,18 @@ A typical entry looks like this:
...
+6. If the above steps does not result in a value, and if there is a fallback field in the legacy data, mapped by the *fallback_legacy_field*, this field will be returned
```
Comment (bltravis) [inline docs/mapping_file_based_mapping.md:62]: "…above steps do…"

File: docs/mapping_file_based_mapping.md [hunk 5]
```diff
@@ -52,6 +52,18 @@ A typical entry looks like this:
...
+7. If none of the above have resulted in a value, and there is an entry for *fallback_value* in the mapping entry, this value will be returned
```
Comment (bltravis) [inline docs/mapping_file_based_mapping.md:63]: "If there is still no value, and there is an entry for…"

File: src/folio_migration_tools/mapping_file_transformation/mapping_file_mapping_base_impl.py [hunk 6]
```diff
@@ -29,12 +28,4 @@ def __init__(
...
-        legacy_item_keys = self.mapped_from_legacy_data.get(folio_prop_name, [])
-
-        if len(legacy_item_keys) == 1 and folio_prop_name in self.mapped_from_values:
-            return self.mapped_from_values.get(folio_prop_name, "")
-
-        legacy_values = MappingFileMapperBase.get_legacy_vals(legacy_item, legacy_item_keys)
-        if len(legacy_item_keys) > 1:
-            self.migration_report.add(Blurbs.Details, f"{legacy_item_keys} were concatenated")
-        return " ".join(legacy_values).strip()
+        return super().get_prop(legacy_item, folio_prop_name, index_or_id)
```
Comment (bltravis) [inline src/folio_migration_tools/mapping_file_transformation/mapping_file_mapping_base_impl.py:31]: Curious why this method is still here if all it does is call the same method on the parent class?

File: src/folio_migration_tools/mapping_file_transformation/notes_mapper.py [hunk 7]
```diff
@@ -69,22 +70,7 @@ def map_notes(self, legacy_object, legacy_id, object_uuid: str, record_type: FOL
...
-                folio_prop_name, self.record_map["data"]
-            )
-        )
-        if len(map_entries) > 1:
-            self.migration_report.add(Blurbs.Details, f"{legacy_item_keys} were concatenated")
-        return " ".join(
-            MappingFileMapperBase.get_legacy_value(legacy_item, map_entry, self.migration_report)
-            for map_entry in map_entries
-        ).strip()
+        return super().get_prop(legacy_item, folio_prop_name, index_or_id)
```
Comment (bltravis) [inline src/folio_migration_tools/mapping_file_transformation/notes_mapper.py:73]: Same question as in the previous file. Why is this method defined if all it does is call the same method on the parent class? (admit, I could be missing something…)

**Review by bltravis — APPROVED** [review]
I didn't see anything that needed to be addressed before this could be merged. There are some questions and wording suggestions for the docs portion.

File: src/folio_migration_tools/mapping_file_transformation/mapping_file_mapping_base_impl.py (hunk 6 above)
Comment (fontanka16) [reply src/folio_migration_tools/mapping_file_transformation/mapping_file_mapping_base_impl.py:31]: Good point. It is just confusing. The point of the class was to test a "clean" implementation of get_prop.

File: src/folio_migration_tools/mapping_file_transformation/notes_mapper.py (hunk 7 above)
Comment (fontanka16) [reply src/folio_migration_tools/mapping_file_transformation/notes_mapper.py:73]: Equally good point. I cleaned the previous implementation out before, and the logical thing to do is to remove them entirely

**Pull request merged**